controllers/vspherecluster_controller.go

```diff
@@ -296,11 +328,39 @@ func (r clusterReconciler) reconcileAPIEndpoints(ctx *context.ClusterContext) er
 		// for this VSphereCluster into the analogous CAPI Cluster using an
 		// UnstructuredReader.
 		ctx.VSphereCluster.Status.APIEndpoints = []infrav1.APIEndpoint{apiEndpoint}
+		vsphereCluster := ctx.VSphereCluster.DeepCopy()
+
+		// Enqueue a reconcile request for the cluster when the target API
+		// server is online.
+		go func() {
+			for {
+				if r.isAPIServerOnline(ctx) {
+					break
+				}
+				time.Sleep(time.Second * 1)
+			}
+			ctx.Logger.Info("triggering GenericEvent", "reason", "api-server-online")
+			eventChannel := ctx.GetGenericEventChannelFor(vsphereCluster.GetObjectKind().GroupVersionKind())
+			eventChannel <- event.GenericEvent{
+				Meta:   vsphereCluster,
+				Object: vsphereCluster,
+			}
+		}()
+
 		return nil
 	}
 	return infrautilv1.ErrNoMachineIPAddr
 }
 
+func (r clusterReconciler) isAPIServerOnline(ctx *context.ClusterContext) bool {
+	if client, err := infrautilv1.NewKubeClient(ctx, ctx.Client, ctx.Cluster); err == nil {
```

Comment:
Alternatively, you can use `ctx.Client` to get a `corev1.NodeList` so you don't have to create a new clientset here.

Reply:
Hmm. Perhaps I don't understand? In this bit of code `ctx.Client` is for the management server and we're getting a client to the target server to validate that it is online.

Reply:
My bad. Misinterpreted what this is doing.